packages/core/test/acceptance/di_spec.ts

```diff
@@ -739,6 +716,742 @@ describe('di', () => {
+               const expectedErrorMessage = ivyEnabled ?
+                   /NG0201: No provider for Injector found in NodeInjector/ :
+                   /No provider for Injector/;
+               expect(() => TestBed.createComponent(ParentComponent))
+                   .not.toThrowError(expectedErrorMessage);
+             });
+        });
+
+        describe('ElementRef', () => {
+          it('should lookup module injector in case @SkipSelf is used for `ElementRef` token and Component has no parent',
+             () => {
+               let componentElement: ElementRef;
+               let moduleElement: ElementRef;
+               @Component({template: '<div>component</div>'})
+               class MyComponent {
+                 constructor(@SkipSelf() public el: ElementRef) {
+                   componentElement = el;
+                 }
+               }
+
+               @NgModule({
+                 declarations: [MyComponent],
+                 providers: [{
+                   provide: ElementRef,
+                   useValue: {from: 'NG_MODULE'},
+                 }]
+               })
+               class MyModule {
+                 constructor(public el: ElementRef) {
+                   moduleElement = el;
+                 }
+               }
+
+               TestBed.configureTestingModule({
+                 imports: [MyModule],
+               });
+               const fixture = TestBed.createComponent(MyComponent);
+               fixture.detectChanges();
+
+               expect((moduleElement! as any).from).toBe('NG_MODULE');
+               expect((componentElement! as any).from).toBe('NG_MODULE');
+             });
+
+          it('should return host node when @SkipSelf is used for `ElementRef` token and Component has no parent node',
+             () => {
+               let parentElement: ElementRef;
+               let componentElement: ElementRef;
+               @Component({selector: 'child', template: '...'})
+               class MyComponent {
+                 constructor(@SkipSelf() public el: ElementRef) {
+                   componentElement = el;
+                 }
+               }
+
+               @Component({
+                 template: '<child></child>',
+               })
+               class ParentComponent {
+                 constructor(public el: ElementRef) {
+                   parentElement = el;
+                 }
+               }
+
+               TestBed.configureTestingModule({
+                 imports: [CommonModule],
+                 declarations: [ParentComponent, MyComponent],
+               });
+               const fixture = TestBed.createComponent(ParentComponent);
+               fixture.detectChanges();
+
+               expect(componentElement!).toEqual(parentElement!);
+             });
+
+          it('should support @SkipSelf when injecting ElementRef', () => {
+            let parentRef: ElementRef|undefined;
+            let childRef: ElementRef|undefined;
```

Comment:
Do we need `|undefined` here?
```suggestion
            let parentRef: ElementRef;
            let childRef: ElementRef;
```

Reply:
Nope. Removed.